metaci/repository/tests/test_utils.py

```diff
@@ -0,0 +1,191 @@
+from unittest import mock
+
+import pytest
+
+from metaci.build.exceptions import BuildError
+from metaci.conftest import BuildFactory
+from metaci.repository import utils
+
+
+def setup_build_with_status(status):
+    """Sets items that need to be mocked"""
+    build = BuildFactory(status=status)
+    assert build.get_status() == status
+
+    # needed for metaci.repository.utils.create_status()
+    repo = mock.Mock()
+    repo.create_status = mock.Mock()
+    build.plan.context = mock.Mock()
+    build.repo.get_github_api = mock.Mock(return_value=repo)
+    return build, repo
+
+
+@pytest.mark.django_db
+def test_create_status__queued():
+    build, repo = setup_build_with_status("queued")
+
+    with mock.patch("metaci.build.models.settings.SITE_URL", "initech.co"):
```

Comment:
Instead of doing this separately for each test, you can make a fixture that is automatically initialized once before all tests in the module, and torn down after they're all done:

    @pytest.fixture(scope="module", autouse=True)
    def site_url():
        with mock.patch("metaci.build.models.settings.SITE_URL", "initech.co"):
            yield

(see https://docs.pytest.org/en/stable/fixture.html#autouse-fixtures-xunit-setup-on-steroids)